src/DocSystem.jl

```diff
@@ -31,10 +31,10 @@ binding(any::Any) = throw(ArgumentError("cannot convert `$(repr(any))` to a `Bin
 # The simple definitions.
 #
 binding(b::Docs.Binding) = binding(b.mod, b.var)
-binding(d::DataType)     = binding(d.name.module, d.name.name)
+binding(d::DataType)     = binding(parentmodule(d), nameof(d))
 binding(m::Module)       = binding(m, nameof(m))
 binding(s::Symbol)       = binding(Main, s)
-binding(f::Function)     = binding(typeof(f).name.module, typeof(f).name.mt.name)
+binding(f::Function)     = binding(parentmodule(f), typeof(f).name.mt.name)
```

Comment:
Also `nameof` here?

```suggestion
binding(f::Function)     = binding(parentmodule(f), nameof(f))
```

Reply:
This is not the same thing, I think? `nameof` is equivalent to `.name.name`, but there's an extra `.mt` here

Reply:
I'm a bit foggy on the details since it's been a while since I look at this code, but I believe it should be doing the equivalent thing: https://github.com/JuliaLang/julia/blob/master/base/reflection.jl#L1531-L1546

Reply:
Oh indeed, this should work